src/browserlib/extract-events.mjs

```diff
@@ -0,0 +1,278 @@
+    if (!acc[b[0]]) acc[b[0]] = [];
+    acc[b[0]].push(b[1]);
+    return acc;
+  }, {});
+
+
+  function fromEventElementToTargetInterfaces(eventEl) {
+    if (!eventEl) return;
+
+    if (eventEl.dataset?.dfnFor || eventEl.dataset?.linkFor) {
+      return (eventEl.dataset.dfnFor || eventEl.dataset.linkFor).split(",").map(t => t.trim());
+    } else if (eventEl.getAttribute("href")?.startsWith("#")) {
+      const dfn = document.getElementById(eventEl.getAttribute("href").slice(1));
+      if (dfn && dfn.dataset?.dfnFor) {
+	return dfn.dataset.dfnFor.split(",").map(t => t.trim());
+      }
+    } else if (handledEventNames[eventEl.textContent]?.length) {
+      // Search for on<event> EventHandler in IDL
+    const matchingInterfaces = handledEventNames[eventEl.textContent];
+      if (matchingInterfaces.length === 1) {
+	// only one such handler, we assume it's a match
+	return matchingInterfaces;
+      } else {
+	console.error("[reffy] Multiple event handler named " + eventEl.textContent + ", cannot associate reliably to an interface in " + spec.title);
+      }
+    }
+  }
+
+
+  let events = [];
+  // Look for event summary tables
+  // ignore DOM spec which uses a matching table format
+  // to map to legacy event types
+  let hasStructuredData = false;
+  if (spec.shortname !== "dom") {
+    document.querySelectorAll("table").forEach(table => {
+      const firstHeading = table.querySelector("thead tr th")?.textContent?.trim();
+      if (firstHeading && firstHeading.match(/^Event/) && firstHeading !== "Event handler") {
+	hasStructuredData = true;
+	// Useful e.g. for pointerevents
+	const bubblingInfoColumn = [...table.querySelectorAll("thead th")].findIndex(n => n.textContent.trim().match(/^bubbl/i));
+	const interfaceColumn = [...table.querySelectorAll("thead th")].findIndex(n => n.textContent.trim().match(/^interface/i));
+	table.querySelectorAll("tbody tr").forEach(tr => {
+	  const event = {};
+	  const eventEl = tr.querySelector("*:first-child");
+	  const annotations = eventEl.querySelectorAll("aside, .mdn-anno");
+	  annotations.forEach(n => n.remove());
+
+	  let el = eventEl.querySelector("dfn,a");
+	  if (el && el.tagName === "A" && el.getAttribute("href").startsWith("https:")) {
+	    // we skip when we hit a link pointing to an external spec
+	    // (this is needed since the HTML spec table includes
+	    // links to pointer events)
+	    return;
+	  }
+	  if (!el) {
+	    el = eventEl.querySelector("code");
+	  }
+	  if (el) {
+	    if (el.tagName === "DFN" && el.id) {
+	      event.href = href(el);
+	    } else if (el.tagName === "A") {
+	      event.href = href(document.getElementById(el.getAttribute("href").slice(1)));
+	    }
+	    event.src = { format: "summary table", href: href(el.closest('*[id]')) };
```

Comment:
Here and below, I'm wondering about attaching where the info came from to the event object that we create because the `href` is going to be approximate in a number of cases (due to the use of `closest`), and less stable, at least is specs that still evolve, because fragment is often going to end up with an occurrence index as in `#ref-for-index-term-fire-an-event-8`.

I don't necessarily have anything better to suggest though. That still seems useful info for debugging and improving extraction of events.

Reply:
yeah, I've been using this mostly for debugging; I'm not sure if the `src` object should be there at all in the published data; the `event.href` one seems more broadly relevant, but maybe we should only publish it when we have good stable data (e.g. a `<dfn id>`).

But in general, I think this may be more a data publication matter than a data extraction one.